tools/sslsniff.py

```diff
@@ -94,47 +130,79 @@
         u64 pid_tgid = bpf_get_current_pid_tgid();
         u32 pid = pid_tgid >> 32;
         u32 tid = (u32)pid_tgid;
+        u32 uid = bpf_get_current_uid_gid();
 
-        FILTER
+        PID_FILTER
+        UID_FILTER
 
         bufs.update(&tid, (u64*)&buf);
         return 0;
 }
 
 int probe_SSL_read_exit(struct pt_regs *ctx, void *ssl, void *buf, int num) {
+        u32 zero = 0;
         u64 pid_tgid = bpf_get_current_pid_tgid();
         u32 pid = pid_tgid >> 32;
         u32 tid = (u32)pid_tgid;
+        u32 uid = bpf_get_current_uid_gid();
+        int ret;
 
-        FILTER
+        PID_FILTER
+        UID_FILTER
 
         u64 *bufp = bufs.lookup(&tid);
         if (bufp == 0) {
                 return 0;
         }
 
-        struct probe_SSL_data_t __data = {0};
-        __data.timestamp_ns = bpf_ktime_get_ns();
-        __data.pid = pid;
-        __data.len = PT_REGS_RC(ctx);
+        int len = PT_REGS_RC(ctx);
+        if (len <= 0) // read failed
+            return 0;
+
+        struct probe_SSL_data_t *data = ssl_data.lookup(&zero);
+        if (!data)
+            return 0;
+
+        data->timestamp_ns = bpf_ktime_get_ns();
+        data->pid = pid;
+        data->tid = tid;
+        data->uid = uid;
+        data->len = (u32)len;
+        data->buf_filled = 0;
+        u32 buf_copy_size = min((size_t)MAX_BUF_SIZE, (size_t)len);
 
-        bpf_get_current_comm(&__data.comm, sizeof(__data.comm));
+        bpf_get_current_comm(&data->comm, sizeof(data->comm));
 
         if (bufp != 0) {
-                bpf_probe_read_user(&__data.v0, sizeof(__data.v0), (char *)*bufp);
+                ret = bpf_probe_read_user(&data->buf, buf_copy_size, (char *)*bufp);
         }
 
         bufs.delete(&tid);
 
-        perf_SSL_read.perf_submit(ctx, &__data, sizeof(__data));
+        if (!ret) {
+            data->buf_filled = 1;
+        } else {
+            buf_copy_size = 0;
+        }
+
+        perf_SSL_read.perf_submit(ctx, data, EVENT_SIZE(buf_copy_size));
         return 0;
 }
 """
 
+TASK_COMM_LEN = 16  # linux/sched.h
```

Comment:
This `TASK_COMM_LEN` is unused.

Reply:
You absolutely right, but it was the case even before my change :smile: . Anyway, I'll remove it.